### PR TITLE
[FLINK-3821] Reduce Guava usage in flink-java

### PR DESCRIPTION
**Pull request by zentol**
Removes all Guava usages in non-test files.

-replaced CharSets with StandardCharsets
-added checkElementIndex() method to Flink Preconditions
-replaced Guava Preconditions with Flink Preconditions
-removed single usages of Ints.max() and Joiner()

### CODE REVIEW

File: flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java [hunk 1]
```diff
@@ -133,13 +132,17 @@ public OUT readRecord(OUT reuse, byte[] bytes, int offset, int numBytes) throws
 	protected static boolean[] toBooleanMask(int[] sourceFieldIndices) {
 		Preconditions.checkNotNull(sourceFieldIndices);
 
+		int max = 0;
 		for (int i : sourceFieldIndices) {
 			if (i < 0) {
 				throw new IllegalArgumentException("Field indices must not be smaller than zero.");
 			}
+			if (i > max) {
```
Comment (greghogan) [inline flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java]: Use the intrinsic `max = Math.max(max, i);`?

File: flink-java/src/main/java/org/apache/flink/api/java/utils/DataSetUtils.java [hunk 2]
```diff
@@ -104,7 +104,10 @@ public void open(Configuration parameters) throws Exception {
 							@Override
 							public List<Tuple2<Integer, Long>> initializeBroadcastVariable(Iterable<Tuple2<Integer, Long>> data) {
 								// sort the list by task id to calculate the correct offset
-								List<Tuple2<Integer, Long>> sortedData = Lists.newArrayList(data);
+								List<Tuple2<Integer, Long>> sortedData = new ArrayList();
```
Comment (greghogan) [inline flink-java/src/main/java/org/apache/flink/api/java/utils/DataSetUtils.java]: Add the diamond operator to ArrayList?

**Comment by greghogan** [comment]
+1 with two comments.

**Comment by zentol** [comment]
@greghogan Thanks for looking over it, I've addressed your comments.

**Comment by fhueske** [comment]
Thanks @zentol. Will merge this PR

**Pull request closed without merging**